### ISSUE TITLE
Allow DISTCC_PREFIX to use DistCC with other compiler wrappers such as CCache.

### ISSUE THREAD

**Issue by Salamandar** [distcc/distcc#230]
Reading the environment variable DISTCC_PREFIX would most easily allow DistCC to call CCache or any other compiler wrapper that is installed server-side.
This variable, client-side, should not change anything.

**Comment by afbjorklund** [comment]
Curently the distcc documentation ([distccd.1](https://github.com/distcc/distcc/blob/master/man/distccd.1#L324)) says this:

>        Because ccache does not cache compilation from .i files,
>        it is not useful to call it from distccd.

That seems to be a bit hard, and could be edited.

Using ccache in this manner avoids the "direct mode" and means that the source files will _always_ have to be preprocessed (and then sent to the server, that has to send the object files etc. back to the client). But if you have a lot of clients, that are _not_ sharing their compiler cache, then it could maybe be useful. Not sure what other compiler wrappers would be involved on the server, but maybe there is something out there ?

BTW the reason that direct mode fails, is because the client source files are missing on the distcc server

And, yeah, it would be better to share the cache...

**Comment by Salamandar** [comment]
Yup, I read some of the DistCC source code and tried to implement this but no success (command failed… But dunno why).

What you say about preproc… Yeah, but aren't source files *always* preprocessed on the client-side ?

**Comment by afbjorklund** [comment]
> Yeah, but aren't source files always preprocessed on the client-side ?

In "direct mode", files do not have to preprocessed (only checksummed)

**Comment by digitalist** [comment]
Doesn't setting ccache prefix to distcc covers that case?
